### PR TITLE
executor: support extractor for mem table region status

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33348

Problem Summary:

Support push down conditions `table_id` in selection for `tikv_region_status` memory table in order to avoid fetch all region status each time which may cause tidb OOM during importing data by lightning

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)

As unit test couldn't test result due to having no pd, I manually tested result.

```sql
mysql> select * from information_schema.TIKV_REGION_STATUS where table_id = 19;
+-----------+--------------------------------------+--------------------------------------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
| REGION_ID | START_KEY                            | END_KEY                              | TABLE_ID | DB_NAME | TABLE_NAME | IS_INDEX | INDEX_ID | INDEX_NAME | EPOCH_CONF_VER | EPOCH_VERSION | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE | APPROXIMATE_KEYS | REPLICATIONSTATUS_STATE | REPLICATIONSTATUS_STATEID |
+-----------+--------------------------------------+--------------------------------------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
|        20 | 7480000000000000FF1300000000000000F8 | 7480000000000000FF1500000000000000F8 |       19 | mysql   | help_topic |        1 |        1 | name       |              1 |            10 |             0 |          0 |                1 |                0 | NULL                    |                      NULL |
|        20 | 7480000000000000FF1300000000000000F8 | 7480000000000000FF1500000000000000F8 |       19 | mysql   | help_topic |        0 |     NULL | NULL       |              1 |            10 |             0 |          0 |                1 |                0 | NULL                    |                      NULL |
+-----------+--------------------------------------+--------------------------------------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
2 rows in set (0.01 sec)

mysql> explain select * from information_schema.TIKV_REGION_STATUS where table_id = 19;
+----------------+----------+------+--------------------------+------------------+
| id             | estRows  | task | access object            | operator info    |
+----------------+----------+------+--------------------------+------------------+
| MemTableScan_5 | 10000.00 | root | table:TIKV_REGION_STATUS | table_id in {19} |
+----------------+----------+------+--------------------------+------------------+
1 row in set (0.00 sec)

mysql> explain select * from information_schema.TIKV_REGION_STATUS where table_id >18 and table_id < 20;
+----------------------+----------+------+--------------------------+------------------------------------+
| id                   | estRows  | task | access object            | operator info                      |
+----------------------+----------+------+--------------------------+------------------------------------+
| Selection_5          | 8000.00  | root |                          | gt(Column#4, 18), lt(Column#4, 20) |
| └─MemTableScan_6     | 10000.00 | root | table:TIKV_REGION_STATUS |                                    |
+----------------------+----------+------+--------------------------+------------------------------------+
2 rows in set (0.00 sec)

mysql> select * from information_schema.TIKV_REGION_STATUS where table_id >18 and table_id < 20;
+-----------+--------------------------------------+--------------------------------------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
| REGION_ID | START_KEY                            | END_KEY                              | TABLE_ID | DB_NAME | TABLE_NAME | IS_INDEX | INDEX_ID | INDEX_NAME | EPOCH_CONF_VER | EPOCH_VERSION | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE | APPROXIMATE_KEYS | REPLICATIONSTATUS_STATE | REPLICATIONSTATUS_STATEID |
+-----------+--------------------------------------+--------------------------------------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
|        20 | 7480000000000000FF1300000000000000F8 | 7480000000000000FF1500000000000000F8 |       19 | mysql   | help_topic |        1 |        1 | name       |              1 |            10 |             0 |          0 |                1 |                0 | NULL                    |                      NULL |
|        20 | 7480000000000000FF1300000000000000F8 | 7480000000000000FF1500000000000000F8 |       19 | mysql   | help_topic |        0 |     NULL | NULL       |              1 |            10 |             0 |          0 |                1 |                0 | NULL                    |                      NULL |
+-----------+--------------------------------------+--------------------------------------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
2 rows in set (0.02 sec)
```

### Release note

```release-note
None
```

### CODE REVIEW

**Review by rebelice — APPROVED** [review]

File: planner/core/memtable_predicate_extractor.go [hunk 1]
```diff
@@ -1555,3 +1555,52 @@ func (e *ColumnsTableExtractor) explainInfo(p *PhysicalMemTable) string {
...
+	remained, _, tableIDSet := e.extractCol(schema, names, predicates, "table_id", true)
+	if tableIDSet.Count() < 1 {
+		return predicates
+	}
+	var tableID int64
+	var err error
+	for key := range tableIDSet {
+		tableID, err = strconv.ParseInt(key, 10, 64)
+		if err != nil {
+			e.tablesID = nil
```
Comment (winoros) [inline planner/core/memtable_predicate_extractor.go:1580]: Do we need to log the error msg?

Comment (Yisaer) [reply planner/core/memtable_predicate_extractor.go:1583]: updated.

**Review by winoros — APPROVED** [review]

**Pull request merged**